File: src/model/vote.ts
export interface Vote{
    coinId : string;
    voteType : VoteEnum;
    votePercentage : number;
}

export const enum VoteEnum{
    UP,
    DOWN
}

export interface Votes{
    votes: Vote[];
    upVotes: number;
    downVotes: number;
    totalVotes: number;
}

File: Back-end/src/main/java/com/bitvote/project/vote/Vote.java
package com.bitvote.project.vote;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Builder
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Vote {
    @Id
    @GeneratedValue
    private Long id;
    private String coinId;
    private VoteEnum voteType;

    @Max(100)
    @Min(0)
    private Double votePercentage;
}
